packages/web/src/components/composer-input/ComposerInput.tsx

```diff
@@ -188,18 +189,26 @@ export const ComposerInput = (props: ComposerInputProps) => {
     async (e: ChangeEvent<HTMLTextAreaElement>) => {
       setValue(e.target.value)
       const editedValue = await resolveLinks(e.target.value)
+      if (submittedRef.current) {
+        return
+      }
       setValue(editedValue)
       // TODO: Need to update this to move to the proper position affect link change to human text
       // setTimeout(() => {
       //   textarea.selectionStart = cursorPosition
       //   textarea.selectionEnd = cursorPosition
       // }, 0)
     },
-    [resolveLinks, setValue]
+    [resolveLinks, setValue, submittedRef]
   )
 
   const handleSubmit = useCallback(() => {
+    submittedRef.current = true
     onSubmit?.(restoreLinks(value), linkEntities, userMentionIds)
+    // Debounce the submission
+    setTimeout(() => {
+      submittedRef.current = false
+    }, 1000)
```

Comment:
curious why we need to debounce this? and why a setTimeout instead of `useDebounce`?

Reply:
is the idea that you want to prevent the user from submitting then quickly typing some more in the composer?

Reply:
the idea is that you want to "cancel" the async on change handler when submit is pressed and debounce the time it is canceled for. agreed not super great, but not sure of a better solve. ill try a bit more and add a comment if I can't come up with anything else?

Reply:
got a better solve with an addl ref. will merge now & can do any other feedback async!